### PR TITLE
Avoid KeyError on event without data field

### PR DESCRIPTION
**Pull request by Rubensei**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 def readme():
-    with open('README.md') as f:
+    with open('README.md', encoding="utf8") as f:
```
Comment (deepbrook) [inline setup.py:9]: That's an unnecessary change, imo :thinking:

Comment (Rubensei) [reply setup.py:9]: Was giving me error because sometimes python tried opening it with other encoding while installing, don't even know why tbh

**Comment by Rubensei** [comment]
Just noticed this isn't actually an issue on official pusher server, it's on the pusher custom implementation i'm using (beyondcode/laravel-websockets) 🤦‍♂

So... not strictly necessary to adapt pysher to it but it's still better practice to use get() (?)

File: setup.py (hunk 1 above)
Comment (deepbrook) [reply setup.py:9]: Ach, I also misread the change - I thought you were removing it. 
Adding it is fine - I've gotten into the habit of always explicitly stating the encoding since I wrote this code.

**Comment by deepbrook** [comment]
> Just noticed this isn't actually an issue on official pusher server, it's on the pusher custom implementation i'm using (beyondcode/laravel-websockets) man_facepalming

No worries :smile: it happens.

> 
> So... not strictly necessary to adapt pysher to it but it's still better practice to use get() (?)

hm, I'd say it depends. In principle, I'd say it *should* throw an error, since the key is expected to be there. Its absence shouldn't have to be handled by the downstream function (which is what returning `None` from `dict.get()` would do). On the other hand, it shouldn't just throw a `KeyError` for custom implementations like the one you're using.

Initially, I was thinking of catching the `KeyError` and calling the function anyway if it's a `pong` request, but again - that seems too dependent on the custom implementation.

In the very least, it should perhaps raise a better exception.

**Comment by Rubensei** [comment]
Thing is pusher is vague on their documentation about data field:

[Here](https://pusher.com/docs/channels/library_auth_reference/pusher-websockets-protocol#detecting-that-the-connection-is-alive) you can see that ping & pong messages should have an empty data field
> This means that Pusher Channels will respond to a WebSocket protocol ping message with a pong message, and also it will respond to apusher:ping event with a pusher:pong event **(both have empty data)**.

On same doc page you can [later](https://pusher.com/docs/channels/library_auth_reference/pusher-websockets-protocol#double-encoding) see them telling data field is optional:
> All events received and sent by clients **can contain** a data field.

Edit: If pusher allows event data field to be optional the checks should be handled by the functions bound to those events 🤔

**Comment by deepbrook** [comment]
> Edit: If pusher allows event data field to be optional the checks should be handled by the functions bound to those events thinking

That's true. Let's merge it then. :)
